docs/guide/quickstart.rst

```diff
@@ -0,0 +1,298 @@
+    We don't recommend Python's built in IDLE, as the lack of features compared to other simple editors makes it very bothersome for projects with many files.
+
+3. Now you can start creating your bot. The following steps will go over a simple bot line-by-line to help you understand what's happening.
+
+First, you need to import the discord.py library:
+
+.. code-block:: python3
+
+    import discord
+    from discord.ext import commands
+
+We import ``discord.ext.commands`` here as we will need it for our bot.
+
+.. code-block:: python3
+
+    intents = discord.Intents.default()
+    intents.message_content = True
+    bot = commands.Bot(command_prefix="!", description="This is my awesome bot!", intents=intents)
+
+This is your bot instance. You can name the variable whatever you like but for simplicity's sake we will name it ``bot``.
+You will need to specify a ``command_prefix`` here, we use ``!``, but you can use any string, or a list of strings for multiple prefixes.
+We also add the :attr:`Intents.message_content` intent, so that our bot can read regular text messages. You can read more in the `Intents guide <_guide_intents>`_.
+
+.. note::
+
+    As of 2.0, the ``intents`` argument is required.
+    ``Intents`` are the method to specify which gateway events you wish to receive. ``Intents.default()`` means you will receive all events that are not locked behind bot verification.
+    For more information, see :ref:`intents_primer`.
+
+.. code-block:: python
+
+    @bot.listen()
+    async def on_ready():
+        print(f"Ready! I am {bot.user} and my ID is {bot.user.id}")
+
+This is the ``ready`` event. It is called when the bot has finished loading and everything is cached.
+We use the :meth:`@bot.listen() <ext.commands.Bot.listen>` decorator as to not override the main event.
+For a list of available events, see :ref:`discord-api-events`.
+
+.. _guide_quickstart_bot_event_warning:
+
+.. warning::
+
+    Using :meth:`@bot.event <ext.commands.Bot.event>` is also valid, but keep in mind that using :meth:`@bot.event <ext.commands.Bot.event>` with the ``on_message`` event can cause your commands to stop responding.
+
+    To remedy this, either switch to :meth:`@bot.listen() <ext.commands.Bot.listen>`, or add :meth:`~ext.commands.Bot.process_commands` to your ``on_message`` event:
+
+    .. code-block:: python
+
+        @bot.event
+        async def on_message(message: discord.Message):
+            print(f'Received "{message.clean_content}" from {message.author}')
+            # IMPORTANT:
+            await bot.process_commands(message)
+
+.. warning::
+
+    ``on_ready`` can and will be called multiple times throughout your bot's uptime. You should avoid doing any kind of state-management here, such as connecting and loading your database.
+
+.. code-block:: python
+
+    @bot.command()
+    async def echo(ctx: commands.Context, *, sentence: commands.clean_content):
+        await ctx.send(sentence)
```

Comment:
I don't think this is a particularly good intro command. It does too much from the get go. It's better if you have simpler converters (such as `int`) and simpler parameters. I know it's weird to say considering this uses a single parameter but you have to understand quite a lot from this simple example. 

The `commands.rst` guide actually tries to do it in a digestible order. It would be ideal to link back to it instead of trying to repeat its content.